android/src/main/java/io/sentry/react/RNSentryModuleImpl.java

```diff
@@ -616,6 +621,48 @@ public void disableNativeFramesTracking() {
         }
     }
 
+    public WritableMap startProfiling() {
+        final WritableMap result = new WritableNativeMap();
+        try {
+            HermesSamplingProfiler.enable();
+            result.putBoolean("started", true);
+        } catch (Throwable e) {
+            result.putBoolean("started", false);
+            result.putString("error", e.toString());
+        }
+        return result;
+    }
+
+    public WritableMap stopProfiling() {
+        final WritableMap result = new WritableNativeMap();
+        try {
+            HermesSamplingProfiler.disable();
+
+            final File output = File.createTempFile(
+                "sampling-profiler-trace", ".cpuprofile", reactApplicationContext.getCacheDir());
+
+            if (this.debugEnabled) {
+                logger.log(SentryLevel.INFO, "Profile saved to: " + output.getAbsolutePath());
+            }
+
+            try (final BufferedReader br = new BufferedReader(new FileReader(output));) {
+                HermesSamplingProfiler.dumpSampledTraceToFile(output.getPath());
```

Comment:
Is there a way to avoid the IO and read directly from HermesSamplingProfiler?

Reply:
At the moment the interface has only dumpToFileExported. 
https://github.com/facebook/react-native/blob/4d40cebe4017509098d889fdbb75761e895a21e3/packages/react-native/ReactAndroid/src/main/java/com/facebook/hermes/instrumentation/HermesSamplingProfiler.java#L13-L32

We could make our own wrapper around the C++ Hermes static library, but it would require more patching of the apps to ensure that our cpp code compiles. At the moment we don't have any Cpp dependencies as we use the Java wrapper that is part of RN.

Reply:
Gotcha, lets keep it as it is now, can we open a feature request in the RN repo instead? I believe this would he beneficial for future profilers, it's leas overhead to transform the profile into something i stead dumping to a file.

Reply:
Yes, definitely, a feature request on the RN repo makes sense.